### PR TITLE
LNS: Wallet Value Decrypt and Name Hash RPC Helpers

### PR DESCRIPTION
**Pull request by Doy-lee**
For desktop wallet not needing to import entire crypto library for simple operations.

- Also ensures that we store the address_parse_info as the mapping_value for wallet addresses in the LNS DB. But since we haven't enabled LNS entries for wallets this should not cause any problems on the network (until we force everyone to upgrade on the next hard fork enabling wallets).

- Print out decrypted mapping values properly depending on the mapping type.

- Fix Base32z encode bug using sizeof(std::string)

### CODE REVIEW

**Comment by Doy-lee** [comment]
That would be an issue if it was available on the daemon, but this is for the wallet rpc and I don't think there's such a thing as a public wallet rpc?

**Comment by jagerman** [comment]
Oh crud, I misread that!  In that case then yeah, this looks good.

File: src/common/base32z.h [hunk 1]
```diff
@@ -21,7 +21,7 @@ const char *encode(std::string const &src, stack_t &stack)
   size_t ret = 0, pos = 1;
   int bits     = 8;
   uint32_t tmp = src[0];
-  size_t len   = sizeof(src);
```
Comment (jagerman) [inline src/common/base32z.h]: Ew, ew, ew.  How did this code even work?

Comment (Doy-lee) [reply src/common/base32z.h]: :p

**Pull request merged**